src/java.desktop/share/classes/java/awt/Desktop.java

```diff
@@ -417,7 +405,6 @@ public void open(File file) throws IOException {
      */
     public void edit(File file) throws IOException {
         file = new File(file.getPath());
-        checkExec();
         checkActionSupport(Action.EDIT);
         file.canWrite();
```

Comment:
Not related to this PR, but it looks like that the `file.canWrite();` check is useless here, as its result is ignored.